src/cdk-experimental/menu/menu.md

```diff
@@ -0,0 +1,241 @@
+spec. Standalone menu triggers cannot be grouped together meaning each single trigger is a tabstop.
+Standalone triggers can be opened from the keyboard and the opened menu behaves just like it does in
+the other implementations.
+
+<!-- TODO standalone trigger example (use from above?) -->
+
+### Menu Bars
+
+The `CdkMenuBar` directive follows the [ARIA menubar][menubar] spec and operates similar to a
+desktop app menubar. It consists of at least one `CdkMenuItem` which triggers a submenu. A menubar
+can be layed out horizontally or vertically (defaulting to horizontal). If the style is changed, the
+`orientation` attribute must be set to match in order for the keyboard navigation to work properly
+and for menus to open up in the correct location.
+
+The primary difference between a menubar and a standalone menu is the menubar groups together the
+triggering components. Therefore, menus in a menubar can be opened/closed when hovering between
+their triggers and keyboard navigation listens to left and right arrow buttons which toggles between
+the menu items in the menubar.
+
+<!-- TODO basic menubar example (google docs?) -->
+
+### Context Menus
+
+A context menu is a menu which opens on a right click which occurs within some context component. A
+context is an area of your application which contains some content where you want to open up a menu.
+When a user right-clicks within the context the associated menu opens up next to their cursor - as
+you would expect with a typical right click menu.
+
+<!-- TODO basic context menu example -->
+
+Nested components can have context menu triggers at each level. When a user right clicks, the menu
+associated with the deepest component the user is hovering over is opened up.
+
+```html
+<div [cdkContextMenuTriggerFor]="outer">
+  My outer context
+  <div [cdkContextMenuTriggerFor]="inner">
+    My inner context
+  </div>
+</div>
+```
+
+In the example above, right clicking on "My inner context" will open up the "inner" menu and right
+clicking inside "My outer context" will open up the "outer" menu.
+
+### Inline Menus
+
+Inline menus are groups of menu items which typically do not trigger their own menus and the menu
+items implement some type of on-click behavior. Unlike triggered menus which are opened up from some
+menu item, inline menus do not pop up - they are always visible. The benefit of using an inline menu
+over a group of standalone menu items is that inline menus provide keyboard navigation,
+accessibility and focus management. Menu items within an inline menus are logically grouped
+together.
+
+<!-- TODO inline menu example (gmail buttons?) -->
+
+### Menu Items
+
+Components which interact with or are placed inside of a menu or menubar should typically have the
+`cdkMenuItem` directive provided. This directive allows the items to be navigated to via keyboard
+interaction.
+
+A menuitem by itself can provide some user defined action by hooking into the `cdkMenuItemTriggered`
```

Comment:
Not something I noticed before, but why is there a custom output for this rather than just `click`?

Reply:
I needed a way to hook into the user defined action in order to trigger it from a keyboard action or programatically.